src/laboratory/api/serializers.py

```diff
@@ -348,4 +348,22 @@ def validate(self, value):
         if value['shelf'] is not None:
             if self.laboratory != value['shelf'].furniture.labroom.laboratory:
                 raise ValidationError(detail="Shelf not found on Laboratory")
-        return value
\ No newline at end of file
+        return value
+
+class ShelfObjectModelSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ShelfObject
+        fields = ['id']
+
+class ShelfObjectDeleteSerializer(serializers.Serializer):
+    shelfobj = serializers.PrimaryKeyRelatedField(queryset=ShelfObject.objects.all())
+    def __init__(self, *args, **kwargs):
+        self.laboratory = kwargs.pop('laboratory')
+        super().__init__(*args, **kwargs)
+
+    def validate_shelfobj(self, value):
```

Comment:
If you get the shelf object the way I provided on the other comment you might not need it since it will return a 404 if it doesn't belong... It doesn't mean you need to delete this, you can leave it if you want to be extra sure, but just as a comment.

Reply:
Also, we have been adding the serializers to the file shelfobject/serializers.py, these should be moved there.